From 5ef923d052a0d0b5087f1766af704525d500ab73 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Sat, 10 Oct 2026 08:33:13 +0200
Subject: [PATCH] avformat/vqf: reject more than two channels

Fixes: Timeout
Fixes: xYsvVfSX7ZHq
Fixes: swarm-Integer-overflow-in-VQF-frame_bit_len-computation-3a8153
Found-by: zhang xingxing <2388969553@qq.com>
Suggested-by: zhang xingxing <2388969553@qq.com>

Infinite loop Replicated through UnModified FFmpeg with timeout
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
(cherry picked from commit 5ef923d052a0d0b5087f1766af704525d500ab73)
---
 libavformat/vqf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/vqf.c b/libavformat/vqf.c
index fb668c16da3..a83b59bcdd9 100644
--- a/libavformat/vqf.c
+++ b/libavformat/vqf.c
@@ -152,7 +152,7 @@ static int vqf_read_header(AVFormatContext *s)
             rate_flag           = AV_RB32(comm_chunk + 8);
             avio_skip(s->pb, len-12);
 
-            if (st->codecpar->ch_layout.nb_channels <= 0) {
+            if (st->codecpar->ch_layout.nb_channels <= 0 || st->codecpar->ch_layout.nb_channels > 2) {
                 av_log(s, AV_LOG_ERROR, "Invalid number of channels\n");
                 return AVERROR_INVALIDDATA;
             }
-- 
2.53.0

